pkg/controller/agent/nad/controller.go

```diff
@@ -59,20 +56,15 @@ func (h Handler) OnChange(key string, nad *nadv1.NetworkAttachmentDefinition) (*
 
 	// TODO delete previous vlan id when update nad
 
-	name := os.Getenv(common.KeyNodeName) + "-" + string(networkv1alpha1.NetworkTypeVLAN)
-	nn, err := h.nodeNetworkCache.Get(common.Namespace, name)
-	if err != nil && !apierrors.IsNotFound(err) {
-		return nil, fmt.Errorf("get node network %s failed, error: %w", name, err)
+	v, err := vlan.GetVlan()
+	if err != nil && !errors.As(err, &netlink.LinkNotFoundError{}) && !errors.As(err, &vlan.SlaveNotFound{}) {
```

Comment:
We use errors.As() here in order to distinguish the `LinkNotFoundError` and `SlaveNotFoundError`.  These two errors show that VLAN network has not been configured.

Reply:
Isn't it `errors.Is()`?

Reply:
errors.Is() is used to match error, including the error message inside. Here what we to compare is only the error type.

Reply:
Attach annotation of `netlink.LinkNotFoundError`
> // LinkNotFoundError wraps the various not found errors when
// getting/reading links. This is intended for better error
// handling by dependent code so that "not found error" can
// be distinguished from other errors